kpet/run.py

```diff
@@ -176,11 +176,11 @@ def __get_hosts(database, target, sets):
         pool_suites = []
         for suite in database.suites:
             if suite.matches(target) and \
-               (sets == set() or suite.sets & sets):
+               (suite.sets is None or suite.sets & sets):
```

Comment:
I needed to change it into "suite.sets == sets" because most of the tests don't specify any sets and the "suite.sets & sets" won't work with both of them being "set()".

Reply:
No, that would be wrong, and `suite.sets & sets` would be right. Because a suite (a case) needs to be included if it's a member of at least one set selected from the command line, with the current command-line interface. And `set() & set()` would produce a correct result.

Reply:
set() & set() would produce a set() and bool(set()) is False meaning this will never match anything in most tests

Reply:
So either we add a set to the tests or we have the == there

Reply:
We've moved all "sets" to suite patterns didn't we? And cases are supposed to inherit suite sets, if they don't have any and otherwise have a subset of suite's. This should work, shouldn't it? Am I missing something?

Reply:
I should perhaps clarify I meant replacing the first condition, the & condition is still there

Reply:
`(suite.sets == sets or suite.sets & sets):`

Reply:
Could you write what the complete condition should be in your opinion?

Reply:
Resolved then?

Reply:
Didn't have time for another look yet. Looking now.